src/NuGet.Core/NuGet.Protocol/DependencyInfo/Utils.cs

```diff
@@ -30,52 +32,74 @@ public static VersionRange CreateVersionRange(string stringToParse)
             ILogger log,
             CancellationToken token)
         {
-            var index = await httpSource.GetJObjectAsync(
-                new HttpSourceRequest(registrationUri, log)
-                {
-                    IgnoreNotFounds = true
-                },
-                log,
-                token);
-
-            if (index == null)
+            using (var sourecCacheContext = new SourceCacheContext())
             {
-                // The server returned a 404, the package does not exist
-                return Enumerable.Empty<JObject>();
-            }
+                var httpSourceCacheContext = HttpSourceCacheContext.Create(sourecCacheContext, 0);
 
-            IList<Task<JObject>> rangeTasks = new List<Task<JObject>>();
+                var parts = registrationUri.OriginalString.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+                var packageId = parts[parts.Length - 2];
 
-            foreach (JObject item in index["items"])
-            {
-                var lower = NuGetVersion.Parse(item["lower"].ToString());
-                var upper = NuGetVersion.Parse(item["upper"].ToString());
+                var index = await httpSource.GetAsync(
+                    new HttpSourceCachedRequest(
+                        registrationUri.OriginalString,
+                        $"list_{packageId}_index",
```

Comment:
Should this be `reg_` instead to differentiate it from `list_` used by flat container?